core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClientBuilder.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Builds a new {@link CircuitBreakerClient} or its decorator function.
+ *
+ * @param <T> the type of {@link CircuitBreakerClientBuilder}
+ * @param <U> the type of the {@link Client} that this builder builds or decorates
+ * @param <I> the type of outgoing {@link Request} of the {@link Client}
+ * @param <O> the type of incoming {@link Response} of the {@link Client}
+ */
+public abstract class CircuitBreakerClientBuilder<
+        T extends CircuitBreakerClientBuilder<T, U, I, O>, U extends CircuitBreakerClient<I, O>,
+        I extends Request, O extends Response> {
+
+    @Nullable
+    final CircuitBreakerStrategy strategy;
+
+    @Nullable
+    final CircuitBreakerStrategyWithContent<O> strategyWithContent;
+
+    CircuitBreakerMapping mapping = CircuitBreakerMapping.ofDefault();
```

Comment:
Could you:

- make these 3 fields private and
- add proper package-local getters?

IIRC our inspection profile should warn about this. Let me know it did not warn you about this, so I double-check the inspection profile.

Reply:
Fixed Thank you~!